### PR TITLE
Specification document for badge system

### PR DESCRIPTION
**Pull request by zsck**
See the [relevant issue](https://github.com/hummingbird-me/hummingbird/issues/673)

[Rendered](https://github.com/redwire/hummingbird/blob/badges-spec/docs/badges.md)

**NOTE** This PR does not need to be merged right away.  I am creating it so that we have a place to discuss and increment on the badge system specification.

### CODE REVIEW

**Comment by vevix** [comment]
Great start! :100:. If we're going to store these specs in this repository I would prefer them to be under `docs/` over `doc/`.

**Comment by cybrox** [comment]
Since we have a base to collect all the notes with this, it might be worth going over all the suggestions in the [Old Forum Thread](http://forums.hummingbird.me/t/what-achievements-should-hummingbird-have/6332) again and pick the useful ones out.

**Comment by NuckChorris** [comment]
Would it be better to collect RFCs in issues with a label? That way we can see the markup'd version

**Comment by redwire** [comment]
I think it's a great idea to have an issue for each RFC type of thing.
Having a pull request lying around gives us a lot more information, though. E.g. you can see commits (whose lines can be commented on), and so on.

File: docs/badges.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Hummingbird would like to be able to reward active users for using the site in a way that allows them to
+show off their reputation in the community as well as the variety of anime and dramas they have watched,
+manga they have read, etc.  Hummingbird's badges could be similar in style to 
+[those of MALGraph](https://github.com/rr-/malgraph4/wiki/Achievements).
+
+In this document, we describe the categories of badges available, as well as the specific badges in each
+category, how they are unlocked, and what they represent.  This guide is meant to serve largely as a
+developer-first specification of how badges will be implemented, but in such a way that user documentation
+(in the form e.g. of a blog post) could be easily written.
```
Comment (wopian) [inline docs/badges.md]: (e.g. in the form of a blog post) reads better. "E.g" sounds awkward in the middle of the sentence.

File: docs/badges.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Categories
+
+### Anime
+
+Title | Description | How to Unlock
+------|-------------|--------------
+First Franchise | Completed an entire franchise | Watched every episode of anime in a franchise
```
Comment (wopian) [inline docs/badges.md]: "of **an** anime"

**Comment by NuckChorris** [comment]
That's true, the ability to comment on lines is pretty damn useful. This'll work.

So I agree with the concept so far, seems good. Anyone wanna go through the list of MALgraph badges and see which ones we should steal?

File: docs/badges.md (hunk 2 above)
Comment (redwire) [inline docs/badges.md]: @wopian This one's not referring to just a single anime in a franchise, but all the anime in that franchise.

**Comment by andyx96** [comment]
I can go through all the anime genres from MALgraph, maybe modify them a little bit to cater the hummingbird categorization.

**Comment by vevix** [comment]
Updated the OP to include a rendered file link.

File: docs/badges.md (hunk 2 above)
Comment (wopian) [inline docs/badges.md]: :+1:

**Comment by NuckChorris** [comment]
I'm just merging this into #673 because we don't need two open PRs and an issue just to discuss the spec.  A single issue works fine too.

**Pull request closed without merging**